tools/autograd/templates/VariableType.cpp

```diff
@@ -245,29 +246,24 @@ static void ensure_no_aten_scalars(Tensor & data) {
   }
 }
 
-template<typename T>
-static bool computes_grad_tmpl(T tensors) {
-  if (!GradMode::is_enabled()) {
-    return false;
-  }
-  for (const Tensor& tensor : tensors) {
-    auto& var = static_cast<const Variable&>(tensor);
+struct ComputeRequiresGrad : IterArgs<ComputeRequiresGrad> {
+  bool out = false;
+  using IterArgs<ComputeRequiresGrad>::operator();
+  void operator()(const at::Tensor& tensor) {
+    const auto& var = static_cast<const Variable&>(tensor);
     if (var.defined() && var.requires_grad()) {
-      return true;
+      out = true;
     }
   }
-  return false;
-}
+  bool short_circuit() { return out; }
+};
 
-using TensorRef = std::reference_wrapper<const Tensor>;
-using TensorRefList = std::initializer_list<TensorRef>;
```

Comment:
`reference_wrapper` is no longer needed here, because we don't need to make sure the initializer list in `f({x, y})` constructs references to x and y rather than copies: now it's just `f(x, y)` and the universal reference will do the right thing. (@colesbury)